src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,443 @@
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('keyCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPassword()
+    {
+        return $this->getParameter('keyCertPassword');
+    }
+
+    /**
+     * Password associated with your key certificate.
+     * Can be omitted if a password isn't set.
+     *
+     * @param string $url
+     * @return static
+     */
+    public function setKeyCertPassword($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('keyCertPassword', $value);
+    }
+
+
+    /**
+     * @return string
+     */
+    public function getValidationUrl()
+    {
+        return $this->getParameter('validationUrl');
+    }
+
+    /**
+     * Set the validation URL for the request
+     *
+     * @param string $url
+     * @return static
+     */
+    public function setValidationUrl($url)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('validationUrl', $url);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getMerchantIdentifier()
+    {
+        return $this->getParameter('merchantIdentifier');
+    }
+
+    /**
+     * Set the merchant identifier for the request
+     *
+     * @param string $url
+     * @return static
+     */
+    public function setMerchantIdentifier($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('merchantIdentifier', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getDisplayName()
+    {
+        return $this->getParameter('displayName');
+    }
+
+    /**
+     * Sets the display name for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDisplayName($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('displayName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getApplicationUrl()
+    {
+        /**
+         * @var null|string
+         */
+        return $this->getParameter('applicationUrl');
+    }
+
+    /**
+     * Sets the application url or the "initiativeContext" for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setApplicationUrl($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('applicationUrl', $value);
+    }
+
+    /**
+     * Get HTTP Method.
+     *
+     * This is nearly always POST but can be over-ridden in sub classes.
+     *
+     * @return string
+     */
+    public function getHttpMethod()
+    {
+        return 'POST';
+    }
+
+    /**
+     * @return array
+     */
+    public function getData()
+    {
+        // We must have the validation URL set. If it isn't, don't make the call.
+        $this->validate('validationUrl');
```

Comment:
Should we throw an exception here if the validation URL is not set rather than relying on the user to have set the value before calling `getData()`?

Reply:
That's sort of what this validate function does. It will try to get the passed in data for the request, if that parameter isn't set, it will throw an error.

Reply:
I will update the comment, don't think that was clear either.